engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1976,92 +1965,244 @@ public void storageMigration(final String vmUuid, final StoragePool destPool) {
     private void orchestrateStorageMigration(final String vmUuid, final StoragePool destPool) {
         final VMInstanceVO vm = _vmDao.findByUuid(vmUuid);
 
+        preStorageMigrationStateCheck(destPool, vm);
+
+        try {
+            if(s_logger.isDebugEnabled()) {
+                s_logger.debug(
+                        String.format("Offline migration of %s vm %s with volumes",
+                                vm.getHypervisorType().toString(),
+                                vm.getInstanceName()));
+            }
+
+            migrateThroughHypervisorOrStorage(destPool, vm);
+
+        } catch (ConcurrentOperationException
+                | InsufficientCapacityException // possibly InsufficientVirtualNetworkCapacityException or InsufficientAddressCapacityException
+                | StorageUnavailableException e) {
+            s_logger.debug("Failed to migration: " + e.toString());
+            throw new CloudRuntimeException("Failed to migration: " + e.toString());
```

Comment:
This will break the stack trace. So, what about `throw new CloudRuntimeException(Sring.format ("Failed to migrate VM: %s ", vm.getUuid()), e);` instead?

Reply:
good point however the throwing is old code that was refactored for the multiple catch only. this is how the exception is thrown in the current system.

Reply:
Refactored to use String.format(....)